utils/build-script-impl

```diff
@@ -2012,28 +2116,18 @@ for deployment_target in "${HOST_TARGET}" "${CROSS_COMPILE_TOOLS_DEPLOYMENT_TARG
 done
 # END OF BUILD PHASE
 
+
 # Trap function to print the current test configuration when tests fail.
 # This is a function so the text is not unnecessarily displayed when running -x.
 tests_busted ()
 {
     echo "*** Failed while running tests for $1 $2"
 }
 
-for deployment_target in "${STDLIB_DEPLOYMENT_TARGETS[@]}"; do
-    case ${deployment_target} in
-        linux-* | freebsd-* | macosx-* | cygwin-*)
-            # OK, we can run tests directly.
-            ;;
-        iphoneos-* | iphonesimulator-* | appletvos-* | appletvsimulator-* | watchos-* | watchsimulator-* | android-* )
-            # FIXME: remove this
-            # echo "Don't know how to run tests for ${deployment_target}"
-            continue
-            ;;
-        *)
-            echo "Unknown compiler deployment target: ${deployment_target}"
-            exit 1
-            ;;
-    esac
+for host in "${ALL_HOSTS[@]}"; do
```

Comment:
Aren't the simulator test targets build products of the OSX host? they are in, e.g. `build/Ninja-XXX/swift-macosx-x86_64/test-iphonesimulator-i386`

We want to iterate over every host we built products for, and call `calculate_targets_for_host` to set SWIFT_TEST_TARGETS, and then do the tests with them.

Iterating over `STDLIB_DEPLOYMENT_TARGETS` is incorrect here - those targets will change for every host. Doing this allows us to cross-compile the unit tests which statically link against the runtime.